plotly/offline/offline.py

```diff
@@ -103,76 +170,118 @@ def iplot(figure_or_data, show_link=True, link_text='Export to plot.ly',
         raise ImportError('`iplot` can only run inside an IPython Notebook.')
 
     from IPython.display import HTML, display
-    figure = tools.return_figure_from_figure_or_data(figure_or_data, validate)
 
-    width = figure.get('layout', {}).get('width', '100%')
-    height = figure.get('layout', {}).get('height', 525)
-    try:
-        float(width)
-    except (ValueError, TypeError):
-        pass
-    else:
-        width = str(width) + 'px'
+    display(HTML(_plot_html(
+        figure_or_data, show_link, link_text, validate,
+        '100%', 525)))
 
-    try:
-        float(width)
-    except (ValueError, TypeError):
-        pass
-    else:
-        width = str(width) + 'px'
 
-    plotdivid = uuid.uuid4()
-    jdata = json.dumps(figure.get('data', []), cls=utils.PlotlyJSONEncoder)
-    jlayout = json.dumps(figure.get('layout', {}), cls=utils.PlotlyJSONEncoder)
+def plot(figure_or_data,
+         show_link=True, link_text='Export to plot.ly',
+         validate=True, output_type='file',
+         include_plotlyjs=True,
```

Comment:
In the future, when [plotly.js is modularized](https://github.com/plotly/plotly.js/issues/41), we might allow folks to include sub-bundles here. Maybe something like `plotlyjs_modules=['scatter', 'bar', 'maps']` which would overwrite the `include_plotlyjs` argument? 
cc @etpinard